libs/libc/net/lib_addroute.c

```diff
@@ -47,24 +47,24 @@
  *   netmask  - Network mask defining the external network (required)
  *   router   - Router address that on our network that can forward to the
  *              external network.
+ *   len      - The address struct length.
  *
  * Returned Value:
  *   OK on success; -1 on failure with the errno variable set appropriately.
  *
  ****************************************************************************/
 
-int addroute(int sockfd, FAR struct sockaddr_storage *target,
-             FAR struct sockaddr_storage *netmask,
-             FAR struct sockaddr_storage *router)
+int addroute(int sockfd, FAR void *target, FAR void *netmask,
+             FAR void *router, socklen_t len)
 {
   struct rtentry entry;
 
   /* Set up the rtentry structure */
 
   memset(&entry, 0, sizeof(struct rtentry));
-  memcpy(&entry.rt_dst, target, sizeof(struct sockaddr_storage));
-  memcpy(&entry.rt_genmask, netmask, sizeof(struct sockaddr_storage));
-  memcpy(&entry.rt_gateway, router, sizeof(struct sockaddr_storage));
+  memcpy(&entry.rt_dst, target, len);
+  memcpy(&entry.rt_genmask, netmask, len);
```

Comment:
Any check that `len` is lees or equal to holder size needed?

Reply:
Done